lib/main_menu.rb

```diff
@@ -0,0 +1,56 @@
+require './lib/player'
+require './lib/setup'
+
+class MainMenu
+
+  def initialize
+    @player = Player.new("player")
+  end
+
+  def intro
+    puts ""
+    puts "-----------------------------------------"
+    puts "          Welcome to BATTLESHIP          "
+    puts "Please enter 'p' to play, or 'q' to quit:"
+    puts "-----------------------------------------"
+    print "> "
+    options
+  end
+
+  def options
+    @player.player_input
+    if @player.input == "Q"
+      puts ""
+      exit_game
+    elsif @player.input == 'P'
+      puts ""
+      the_game
+    else
+      puts "Sorry. Invalid entry. Please type 'p' to play or 'q' to quit:\n"
+      print "> "
+      options
+    end
+  end
+
+  def the_game
+    game = Setup.new
+    game.setup
```

Comment:
As a refactor here, let's add a space between the beginning of the loop and the method calls just so we can differentiate what's happening here.